code/starter_script - Copy.R

```diff
@@ -61,14 +61,35 @@ max_temp_timing <- rgnc_by_seg_date %>%
             max_temp_hyb = max(rgcn2_full_temp_c),
             max_timing_hyb = lubridate::yday(date[which.max(rgcn2_full_temp_c)]),
             max_temp_hyb_dat = date[which.max(rgcn2_full_temp_c)],
-            error_obs_proc = abs(max_temp_c - max_temp_proc),
-            error_obs_hyb = abs(max_temp_c - max_temp_hyb),
+            error_temp_obs_proc = abs(max_temp_c - max_temp_proc),
+            error_time_obs_proc = abs(max_timing_tempc - max_timing_proc),
+            error_temp_obs_hyb = abs(max_temp_c - max_temp_hyb),
+            error_time_obs_hub = abs(max_timing_tempc - max_timing_hyb),
             summer_complete = all(170:245 %in% lubridate::yday(date))) %>% 
   filter(summer_complete) %>%
-  mutate(dev_mean_obs_proc = error_obs_proc - mean(error_obs_proc),
-         dev_mean_obs_hyb = error_obs_hyb - mean(error_obs_hyb))
+  mutate(dev_mean_obs_proc = error_temp_obs_proc - mean(error_temp_obs_proc),
+         dev_mean_obs_hyb = error_temp_obs_hyb - mean(error_temp_obs_hyb))
+
+cal_max_temp_timing <- function(observe_data, predict_data, date) {
```

Comment:
Because the max and max timing have a lot of redundant calculations, I suggest combining these two functions, where the function returns both the magnitude and timing metrics.

Reply:
Instead of `date` as an argument, I would call it `doy_test` and set it equal to 170:245. Then, within the function, you would have something like:

`summer_complete = all(doy_test %in% lubridate::yday(date)) `

Reply:
Also, I'm not sure what lines of code you're seeing the error. Can you insert a comment on the line where it's happening, and paste the error output?

Reply:
The code:
 `max_temp_timing_fun <- rgnc_by_seg_date %>% 
  summarize(max_obs_procc = cal_max_temp_timing(temp_c, sntemp_temp_c),
            max_obs_hyb = cal_max_temp_timing(temp_c, rgcn2_full_temp_c))`

Below is the screenshot of the error when I call the function:
![image](https://user-images.githubusercontent.com/65974907/87075572-2d028480-c1e6-11ea-91e3-ffdb8dbbfc43.png)